components/column-linking-table.vue

```diff
@@ -18,23 +19,138 @@
 
 <script>
 
+    // Allows for reference to store data by creating simple, implicit getters
+    // Fields listed in mapState below can be found in the store (index.js)
+    import { mapState } from "vuex";
+
+    // Allows for reference to store actions (index.js)
+    import { mapActions } from "vuex";
+
     export default {
 
         props: {
 
-            categoryClasses: { type: Object, required: true },
-            columnToCategoryMap: { type: Object, required: true },
-            fields: { type: Array, required: true },
-            selectedCategory: { type: String, required: true },
-            tableData: { type: Array, required: true }
+            selectedCategory: { type: String, required: true }
+        },
+
+        data() {
+
+            return {
+
+                columnToCategoryTable: [],
+
+                uiText: {
+
+                    tableFields: [
+
+                        { key: "column" },
+                        { key: "description" }
+                    ]
+                }
+            };
+        },
+
+        created() {
+
+            // Create the data structure for the category to column linking table
+            this.setupColumnToCategoryTable();
+        },
+
+        computed: {
+
+            ...mapState([
+
+                "categoryClasses",
+                "columnToCategoryMap",
+                "dataDictionary",
+                "dataTable"
+            ])
         },
 
         methods: {
 
+            ...mapActions([
+
+                "linkColumnWithCategory",
+                "unlinkColumnFromCategory"
+            ]),
+
             applyCategory(p_row, p_index, p_event) {
 
-                // Tell the parent page that a column has been linked with a category
-                this.$emit("column-name-selected", { column: p_row.column });
+                const dataForStore = { column: p_row.column };
```

Comment:
If I understand correctly here: when I click on a column, this `applyCategory` gets called. And it then tells the store: "hey, someone clicked on this column while this category was active - can you go figure this out?". So I think we can leave it up to the store to figure out whether the meaning of this click is going to be a "unlink" or a "link" action. That is, I think here we should just call the store mutation/action

Reply:
`applyCategory` now calls the action (to be implemented in store refactor) `alterColumnCategoryRelation`. This will take a payload of the column and category and determine whether the column should be linked with this category (if it has no category, or a different category associated with it) or if it should be unlinked (the input category is the same as the one it is currently associated with).

I am leaving in the `emit` for now as I would like to handle the general move of next page access to the layout/store in a separate PR.